New page: aria-owns

estelle opened the pull request:
- [X] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

estelle committed:
New page: aria-owns

ericwbailey commented on files/en-us/web/accessibility/aria/attributes/aria-owns/index.md:
```diff
@@ -0,0 +1,65 @@
+
+Do not use `aria-owns` as a replacement for the DOM hierarchy. If the relationship is represented in the DOM, do not use `aria-owns`.
+
+A child element is owned by it's DOM parent by default: in this case, `aria-owns` should not be used.  While the `aria-owns` attribute can be used to rearrange existing child elements into a different order, this should be avoided. While it may be tempting to order the accessibility layer to match order changes created with the CSS {{CSSXref('order')}} property, avoiding both the `order` property ...
```
```suggestion
A child element is owned by it's DOM parent by default: in this case, `aria-owns` should not be used. Avoid using the `aria-owns` attribute to rearrange existing child elements into a different order. While it may be tempting to order the accessibility layer to match order changes created with the CSS {{CSSXref('order')}} property, avoiding both the `order` property and the `aria-owns` attribute is the best option. 
```

ericwbailey commented on files/en-us/web/accessibility/aria/attributes/aria-owns/index.md:
```diff
@@ -0,0 +1,65 @@
+
+> **Note:** The `aria-owns` attribute should only be used when the relationship cannot be determined from the DOM. 
+
+If an element has both `aria-owns` and DOM children then the order of the child elements is the actual DOM children first, then the elements referenced in `aria-owns`. This order can be changed by including the ID references to the actual DOM children in the `aria-owns` value. 
```
```suggestion
If an element has both `aria-owns` and DOM children, the order of the child elements:

 1. The actual DOM children first, 
 2. Then the elements referenced in `aria-owns`. 
 
 This order can be changed by including the ID references to the actual DOM children in the `aria-owns` value. 
```

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey committed:
Update files/en-us/web/accessibility/aria/attributes/aria-owns/index.md

ericwbailey commented on files/en-us/web/accessibility/aria/attributes/aria-owns/index.md:
```diff
@@ -0,0 +1,70 @@
+
+When elements visually seem to be related but are not associated in the DOM, the `aria-owns` attribute enables creating the relationship that appears on screen in the accessibility layer for use by assistive technology. The **only** reason to include `aria-owns` is to expose a parent/child contextual relationship to assistive technology when the DOM's construction can't provide that relationship. ...
+
+An "owning element" is any DOM ancestor of an element. If an element visually, functionally, or contextually appears to "own" (be an ancestor of) an element, but doesn't actually "own" the element in the DOM, include the `aria-owns` to create that relationship. Add the attribute to the owning element with reference to the non-child owned element (or elements) to tell assistive technology that an  ...
```
@estelle Can we rephrase this paragraph? I think we may use to need synonyms for the word "owns" here, in that it gets confusing determining the difference between inherited ownership and programmatically assigned ownership.

estelle replied:
done

ericwbailey commented on files/en-us/web/accessibility/aria/attributes/aria-owns/index.md:
```diff
@@ -0,0 +1,70 @@
+
+Do not use `aria-owns` as a replacement for the DOM hierarchy. If the relationship is represented in the DOM, do not use `aria-owns`.
+
+A child element is owned by it's DOM parent by default: in this case, `aria-owns` should not be used. Avoid using the `aria-owns` attribute to rearrange existing child elements into a different order. While it may be tempting to order the accessibility layer to match order changes created with the CSS {{CSSXref('order')}} property, avoiding both the `order` property and the `aria-owns` attribute  ...
```
@estelle The mentioning of CSS layout adjustment here is a surprise. Can we rephrase this paragraph, or possibly put up some content earlier to mention that CSS can make modifications or content ordering?

ericwbailey commented on files/en-us/web/accessibility/aria/attributes/aria-owns/index.md:
```diff
@@ -0,0 +1,70 @@
+
+A child element is owned by it's DOM parent by default: in this case, `aria-owns` should not be used. Avoid using the `aria-owns` attribute to rearrange existing child elements into a different order. While it may be tempting to order the accessibility layer to match order changes created with the CSS {{CSSXref('order')}} property, avoiding both the `order` property and the `aria-owns` attribute  ...
+
+When using `aria-owns`, make sure you manage focus order. Ensure the visual focus order matches this assistive technology reading order. 
```
@estelle Do you think we should link to a resource on how to manage focus order here? It might be a new concept for a reader unfamiliar with ARIA/accessibility.

estelle committed:
edits: aria-owns

sideshowbarker commented:
Rebased to fix CI failure (that I caused…)

sideshowbarker committed:
Fix macro syntax (was breaking CI)

ericwbailey reviewed (commented):
One big picture question I'm wondering about that `aria-owns` brings up: should we mention when ARIA is unsuitable for use? 

[VoiceOver on iOS and macOS both do not currently support `aria-owns`](https://a11ysupport.io/tech/aria/aria-owns_attribute), but this is not be apparent to a person reading this documentation. Is it our job to highlight that? I am thinking:

- Yes, because it is very likely a reader won't do the research themselves, or test on a sophisticated suite of AT.
- If yes, how do we stay on top of maintaining this?
- If no, should we have some sort of bare minimum warning message across all ARIA documentation?

estelle commented:
We should put good verbiage to put on every single ARIA page.

estelle committed:
added browser support 

- [`aria-owns` browser support](https://a11ysupport.io/tech/aria/aria-owns_attribute)
